lib/services/recipe_service.dart

```diff
@@ -0,0 +1,22 @@
+
+
+
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+import '../models/recipe.dart';
+
+class RecipeService {
+  static const String url = 'https://hf-android-app.s3-eu-west-1.amazonaws.com/android-test/recipes.json';
+
+  static Future<List<Recipe>> fetchRecipes() async {
+    final response = await http.get(Uri.parse(url));
+
+    if (response.statusCode == 200) {
+      List<dynamic> body = jsonDecode(response.body);
+      List<Recipe> recipes = body.map((dynamic item) => Recipe.fromJson(item)).toList();
+      return recipes;
+    } else {
+      throw Exception('Failed to load recipes');
+    }
+  }
+}
```

Comment:
Should always thinking about writing a testable code.

- Avoid creating static method since it's unable to mock when writing test.
- `body`, `recipes` should be `final`. This values won't changes after created.
-  No need to check statusCode. It will throw the error in catch block if status is not success.
```suggestion
  Future<List<Recipe>> fetchRecipes() async {
    try {
      final response = await http.get(Uri.parse(url));
      final body = jsonDecode(response.body) as List;
      return body.map((dynamic item) => Recipe.fromJson(item)).toList();
    } catch (_) {
      throw Exception('Failed to load recipes');
    }
  }
```